project/src/main/webapp/script.js

```diff
@@ -11,3 +11,51 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/* exported getRecommendation */
+function getRecommendation() {
+  const cuisineType = document.getElementById('cuisine').value;
+  const radius =
+      milesToMeters(parseInt(document.getElementById('distance').value));
+  const priceLevel = document.getElementById('price-level').value;
+  const currLat = document.getElementById('latitude').value;
+  const currLng = document.getElementById('longitude').value;
+  const diningExp = document.getElementById('dining-experience').value;
+  const apiKey = 'AIzaSyBBqtlu5Y3Og7lzC1WI9SFHZr2gJ4iDdTc';
+  const proxyUrl = 'https://cors-anywhere.herokuapp.com/';
+  const textSearchBaseUrl =
+      'https://maps.googleapis.com/maps/api/place/textsearch/json?';
+  const searchParams = new URLSearchParams();
+  searchParams.append('query', cuisineType + ' restaurant');
+  searchParams.append('location', currLat + ',' + currLng);
+  searchParams.append('radius', radius);
+  searchParams.append('key', apiKey);
+  fetch(proxyUrl + textSearchBaseUrl + searchParams)
+      // this will give us the list of restaurants
+      .then((response) => response.json())
+      .then((restaurants) => {
+        fetch('/recommendation', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            restaurants: restaurants.results,
+            cuisineType: cuisineType,
+            lat: currLat,
+            lng: currLng,
+            radius: radius,
+            priceLevel: priceLevel,
+            diningExp: diningExp,
```

Comment:
If you name these all the same as the key you can just do `{restaurants, cuisineTypes, lat, lng, radius, priceLevel, diningExp}`